src/datasets/ground_atom_data.py

```diff
@@ -24,31 +25,35 @@ def create_ground_atom_data(env: BaseEnv, base_dataset: Dataset,
         p: utils.strip_predicate(p)
         for p in annotating_predicates
     }
-    labeleds = {p: 0 for p in annotating_predicates}
-    totals = {p: 0 for p in annotating_predicates}
-    annotations: List[List[Set[GroundAtom]]] = []
-    annotated_preds = set()
-    for traj in base_dataset.trajectories:
-        ground_atoms_traj: List[Set[GroundAtom]] = []
-        for s in traj.states:
-            ground_atoms = sorted(utils.abstract(s, annotating_predicates))
-            subset_atoms = set()
-            if annotating_ratio > 0:
-                for ga in ground_atoms:
-                    pred = ga.predicate
-                    if (totals[pred] == 0 or \
-                            labeleds[pred] / totals[pred] <= annotating_ratio):
-                        # Teacher comments on this atom
-                        stripped_pred = predicates_to_stripped[pred]
-                        stripped_ga = GroundAtom(stripped_pred, ga.objects)
-                        subset_atoms.add(stripped_ga)
-                        labeleds[pred] += 1
-                        annotated_preds.add(pred)
-                    totals[pred] += 1
-            ground_atoms_traj.append(subset_atoms)
-        assert len(traj.states) == len(ground_atoms_traj)
-        annotations.append(ground_atoms_traj)
-    assert len(annotations) == len(base_dataset.trajectories)
-    assert annotated_preds == annotating_predicates, \
-        "Not all predicates are seen in the dataset"
+    # Generate all positive and negative examples
+    pos_examples = DefaultDict(list)  # predicate: [(i, j, ground_atom)]
```

Comment:
I think we generally want to use things imported from `typing` for annotations only. (I don't have a good reason other than convention.) so here, use `from collections import defaultdict`